core/src/main/java/org/apache/druid/guice/LifecycleModule.java

```diff
@@ -37,6 +37,7 @@
  */
 public class LifecycleModule implements Module
 {
+  private final LifecycleScope initScope = new LifecycleScope(Lifecycle.Stage.INIT);
```

Comment:
Can you add a doc somewhere that mentions things in this lifecycle should not log anything. So that means things should expect to fail silently and be fine, or might need to exit the whole jvm if there is a problem. But logging or warning do not fit into this lifecycle

Reply:
At least not during `stop`

Reply:
shutdown runs things in reverse so `INIT` stage runs last which mean the log4j shutdown now currently always runs last instead of near the end, which means it's ok for anything to log up until at least the `INIT` phase shutdown.

Reply:
Oh i misread comment, I see what you are saying, yeah will add.